build/images/base-windows/Dockerfile

```diff
@@ -61,3 +61,32 @@ COPY --from=windows-utility-base ["C:\\\\wins","C:\\\\wins"]
 COPY --from=windows-golang ["C:\\\\Program Files\\\\Go", "C:\\\\Program Files\\\\Go"]
 
 RUN setx /m PATH "C:\Program Files\7-Zip;C:\git\bin;C:\git\usr\bin;C:\mingw64\bin;C:\Program Files\Go\bin;%GOPATH\bin%;%PATH%"
+
+FROM mcr.microsoft.com/windows/servercore:${NANOSERVER_VERSION} as windows-ovs
+ARG WIN_OVS_VERSION
+
+# Switch to a privileged user
+USER ContainerAdministrator
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+ENV WIN_OVS_VERSION ${WIN_OVS_VERSION}
+
+# Install VC
+RUN curl.exe -Lo C:\vc_redist.x64.exe https://aka.ms/vs/17/release/vc_redist.x64.exe; \
+    powershell -Command  "Start-Process -FilePath C:\vc_redist.x64.exe -ArgumentList ‘/install’,’/quiet’,’/norestart’ -Verb RunAs -Wait"
+
+# Download and extract OVS
+RUN curl.exe -fLo C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip https://downloads.antrea.io/ovs/ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip;  \
+    powershell -Command Expand-Archive C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip -DestinationPath C:\  ;  \
+    rm C:\ovs-${env:WIN_OVS_VERSION}-antrea.0-win64.zip
+
+# Download and extract OpenSSL
+RUN curl.exe -Lo C:\openssl-1.0.2u-x64_86-win64.zip https://indy.fulgan.com/SSL/openssl-1.0.2u-x64_86-win64.zip; \
+    powershell -Command Expand-Archive C:\openssl-1.0.2u-x64_86-win64.zip -DestinationPath C:\  ;  \
+    rm C:\openssl-1.0.2u-x64_86-win64.zip
+
+RUN cp C:/libeay32.dll C:/openvswitch/usr/sbin/ ; \
+    cp C:/ssleay32.dll C:/openvswitch/usr/sbin/ ; \
+    mv C:/libeay32.dll C:/openvswitch/usr/bin/ ; \
+    mv C:/ssleay32.dll C:/openvswitch/usr/bin/ ; \
+    mkdir C:/openvswitch/var/run/openvswitch
```

Comment:
ditto, why we need to create this directory in the build image?

Reply:
This folder is required and used by OVS to store runtime files . It contain various files and sockets used for communication between different components of OVS such as ovs-vswitchd, ovsdb-server, and other processes.

Reply:
My question is, OVS doesn't even run in this image, why we need to create it? OVS runs in antrea-windows, not base-windows, right?

Reply:
base-windows is the image used by antrea-windows

Reply:
Hi @rajnkamr are you sure OVS can run in the antrea-windows container with new base image? We didn't find any copy command added in agent image's dockerfile. Can you check if OVS is still running on host?

Reply:
@XinShuYang - the verification is done with base-windows image(not antrea-windows) - rajnishk1608/base-windows:0af6780141 - we will verify this with antrea-windows image, but before that this PR has to get merged and then antrea-windows image will be generated and pushed to dockerhub as antrea-windows generation has failed due to  this error https://github.com/antrea-io/antrea/actions/runs/5242935603/jobs/9466977998?pr=4936

Reply:
@rajnkamr  Since the antrea-windows image has not been verified, I would prefer using a private Docker Hub or adding the related building script to pass CI. Previously, I thought the antrea-windows image had been verified so I asked Quan to upload the base image, it should not be a blocker for the next verification.

Reply:
For testing, you could build base-windows locally, build antrea-windows based on it, then upload it to windows nodes to test. No dockerhub repo is required in the process. And antrea dockerhub repo is not a good place to test patches.

The base-windows dockerfile of this version is not likely the eventual one because the issues I pointed out in the comments. I would suggest to include antrea-windows dockerfile change in this PR so we can clearly know why some files are downloaded in base-windows and how they are going to be used in antrea-windows. You don't need to worry the build job failure. When the PR is approved and tested successfully, I or someone with the dockerhub permission could generate base-image and upload it to make antrea-windows build pass.

Reply:
@XinShuYang @tnqn 
Private build can be created for antrea-windows, we had created it earlier as well during testing, we will verify it and update.

Reply:
@tnqn ,
 Initially we wanted to create a separate image by the name of windows_ovs and then we tested with base_windows for ovs container, we wanted to use either of these images,  hence the changes were done accordingly,

Now If we want to use antrea-windows image then we will have to copy those files and create directory etc in antrea windows dockerfile

Reply:
Antrea windows image is available at rajnishk1608/antrea-windows:latest

Reply:
@tnqn 
Final Antrea-windows image is approx 745 MB, 

since multistage build is used to first download all dependencies and then copy  and then use powershell image receptively in below windows image stages and order
[mcr.microsoft.com/windows/servercore:1809]
[mcr.microsoft.com/windows/nanoserver:1809]
[mcr.microsoft.com/powershell:lts-nanoserver-1809]


As we know when using multiple base images in a Dockerfile, the final image will inherit the filesystem and configuration from the last base image specified. However, it's important to note that the resulting image will not contain the layers from the previous base images. Each base image will only contribute its own layers and modifications to the final image.

So, in this case, if we use the images mcr.microsoft.com/windows/servercore:1809, mcr.microsoft.com/windows/nanoserver:1809, and mcr.microsoft.com/powershell:lts-nanoserver-1809 one after the other in Dockerfile, the final image will have the filesystem and configuration of mcr.microsoft.com/powershell:lts-nanoserver-1809, with any modifications or additions specified in Dockerfile. The layers and configurations of the previous base images will not be present in the final image. so files and exes can be present but any installation will not work.

This is causing issue if we follow this approach as VC application which is used for ovs needs windows sevrecore image to be installed and present in final layer
henc we need to install VC in servercore env 
RUN powershell -Command  "Start-Process -FilePath C:\vc_redist.x64.exe

Reply:
That's why I was saying this PR is not going to work as some steps are not supposed to be executed in the base-windows, and why you should make antrea-windows Dockerfile change in this PR too, to clearly show how the final image is built and why some steps in base-windows is required.

Reply:
I am not strict to merge antrea-ovs and antrea-agent into a signle image if OVS userspace processes are not able to run on base of Windows nano-server. Maybe we could think about using separate images for antrea-ovs (based on servercore ) and antrea-agent (based on powershell:nanoserver) on Windows?

Reply:
BTW, I agree to merge the two or three patches which are focusing on the target of making OVS run in conainer into a single patch for review. The current solution ( more patches in review status ) is difficult to understand the final status.

Reply:
> I am not strict to merge antrea-ovs and antrea-agent into a signle image if OVS userspace processes are not able to run on base of Windows nano-server. Maybe we could think about using separate images for antrea-ovs (based on servercore ) and antrea-agent (based on powershell:nanoserver) on Windows?

OVS userspace processes are not able to run on windows nano-server image. In fact antrea-windows uses powershell nanoserver image, We need to have separate image for ovs based on servercore.

Reply:
> BTW, I agree to merge the two or three patches which are focusing on the target of making OVS run in conainer into a single patch for review. The current solution ( more patches in review status ) is difficult to understand the final status.

I think we can merge 3 patches in two patches one for image and one for other changes, since other patch only need to have image name from this PR .

Reply:
> OVS userspace processes are not able to run on windows nano-server image. In fact antrea-windows uses powershell nanoserver image, We need to have separate image for ovs based on servercore.

@rajnkamr 
As we discussed, another solution is to run an OVS container using an independent OVS image. This approach may allow us to avoid modifying the base and agent image Dockerfiles.

> I think we can merge 3 patches in two patches one for image and one for other changes, since other patch only need to have image name from this PR .

I think we should use only one PR for the convenience of the maintainer's review, you can include multiple commits in the PR, their info can be displayed in [commit](https://github.com/antrea-io/antrea/pull/4936/commits).
 cc @tnqn @wenyingd

Reply:
Description updated with the approach, On windows node, a new windows ovs image will be used for running ovs container, antrea agent container will use antrea-windows image.